### ISSUE TITLE
Ionic build android error

### ISSUE THREAD

**Issue by linrenling** [driftyco/ionic-cli#1096]
:compileArmv7DebugJavaWithJavac 

:compileArmv7DebugNdk UP-TO-DATE
:compileArmv7DebugSources
:transformClassesWithDexForArmv7DebugException in thread "main" java.lang.UnsupportedClassVersionError: com/android/dx/command/Main : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)

Exception in thread "main" java.lang.UnsupportedClassVersionError: com/android/dx/command/Main : Unsupported major.minor version 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)

 FAILED

system information:

Cordova CLI: 6.2.0
Ionic Framework Version: 1.2.4-nightly-1917
Ionic CLI Version: 1.7.15
Ionic App Lib Version: 0.7.2
ios-deploy version: 1.8.5 
ios-sim version: 5.0.6 
OS: Mac OS X El Capitan
Node Version: v5.8.0
Xcode version: Xcode 7.3.1 Build version 7D1014

**Comment by peterrubin** [comment]
Hi
I had the same problem under windows and have solved it so
Download and Update Java SE Development Kit 8u92
And change System Variables JAVA_HOME to C:\Program Files\Java\jdk1.8.0_92\
hope it helps...

**Comment by justinschuldt** [comment]
I solved this problem by changing the Android SDK Build-tools from version 24 to version 23.0.3

In Android Studio I uninstalled Android SDK Build-tools version 24, installed version 23.0.3 and then tried to build the android project with the ionic CLI, and it worked.

**Comment by inductible-dev** [comment]
I had same issue in Cordova/Crosswalk (no Ionic tooling), and resolved as above: changing the Android SDK Build tools from version 24 to version 23.0.3

**Comment by roblav96** [comment]
Does anyone know why using `v24.0.0` causes this error?

**Comment by avenashmosur** [comment]
java.lang.UnsupportedClassVersionError: com/android/dx/command/Main : Unsupported major.minor version 52.0
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
        at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)
Exception in thread "main" 
Error: Error code 1 for command: cmd with args: /s,/c,"C:\Users\mosurav\AndroidStudioProjects\ionicApp\myApp\platforms\android\gradlew cdvBuildDebug -b C:\Users\mosurav\AndroidStudioProjects\ionicApp\myApp\platforms\android\bui
ld.gradle -Dorg.gradle.daemon=true -Pandroid.useDeprecatedNdk=true"


I still have this issue. Changing Build tool version  to 23.0.3 did not help. 
I am using jdk1.7.0_80.
Any idea?

**Comment by ArjunNagraj** [comment]
Uninstalling 24 worked for me too but does anyone know why is it coz of a wrong version in the project folder ?

**Comment by bmnepali** [comment]
Mine worked too on removing the Build Tools 24 and using 23.0.3, i m wondering too why this throws the error!

**Comment by uncooloj** [comment]
Mine worked also thanks alot

**Comment by jgw96** [comment]
Hello! Thanks for using Ionic! As this is an issue with version of the sdk and java that is installed on your machine and not an issue with the Ionic cli I will be closing this issue for now. Thanks for using Ionic!

**Comment by periyasamy17** [comment]
Same problem here..
how to i solve this problem?
when i build fund this error



periyasamy@Periyasamy:~/Desktop/notification-push$ ionic build android
Running command: /home/periyasamy/Desktop/notification-push/hooks/after_prepare/010_add_platform_class.js /home/periyasamy/Desktop/notification-push


add to body class: platform-android

ANDROID_HOME=/opt/android-sdk

JAVA_HOME=/usr/lib/jvm/default-java

Deleting directory /home/periyasamy/.gradle/wrapper/dists/gradle-2.13-all/7hsc6vr6mi3i6i5m7q9hj4ci1q/gradle-2.13

Unzipping /home/periyasamy/.gradle/wrapper/dists/gradle-2.13-all/7hsc6vr6mi3i6i5m7q9hj4ci1q/gradle-2.13-all.zip to /home/periyasamy/.gradle/wrapper/dists/gradle-2.13-all/7hsc6vr6mi3i6i5m7q9hj4ci1q


Exception in thread "main" java.lang.RuntimeException: java.util.zip.ZipException: invalid code lengths set
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:78)
	at org.gradle.wrapper.Install.createDist(Install.java:47)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:48)
Caused by: java.util.zip.ZipException: invalid code lengths set
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.io.FilterInputStream.read(FilterInputStream.java:107)
	at org.gradle.wrapper.Install.copyInputStream(Install.java:189)
	at org.gradle.wrapper.Install.unzip(Install.java:175)
	at org.gradle.wrapper.Install.access$400(Install.java:29)
	at org.gradle.wrapper.Install$1.call(Install.java:70)
	at org.gradle.wrapper.Install$1.call(Install.java:47)
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:65)
	... 3 more

Error: /home/periyasamy/Desktop/notification-push/platforms/android/gradlew: Command failed with exit code 1 Error output:
Exception in thread "main" java.lang.RuntimeException: java.util.zip.ZipException: invalid code lengths set
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:78)
	at org.gradle.wrapper.Install.createDist(Install.java:47)
	at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:129)
	at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:48)
Caused by: java.util.zip.ZipException: invalid code lengths set
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.io.FilterInputStream.read(FilterInputStream.java:107)
	at org.gradle.wrapper.Install.copyInputStream(Install.java:189)
	at org.gradle.wrapper.Install.unzip(Install.java:175)
	at org.gradle.wrapper.Install.access$400(Install.java:29)
	at org.gradle.wrapper.Install$1.call(Install.java:70)
	at org.gradle.wrapper.Install$1.call(Install.java:47)
	at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:65)
	... 3 more

**Comment by rocco** [comment]
For me the solution was to downgrade cordova-plugin-crosswalk-webview from 2.0.0 to 1.8.0. 
2.0.0 discontinues support for Android 4.0 which I needed.

**Comment by maddy619** [comment]
No need to downgrade you sdk version.You can download and install latest java  8(jdk1.8.0_101) for ionic version 2.0.0. then edit your environment variable. Restart cmd > go into your project folder start building. It will be success

**Comment by rocco** [comment]
@maddy619 you're right. I upgraded Java JDK to jdk1.8.0_102 just after my comment and now upgraded SDK tools back to 24.0.2 after your comment - still compiles yay!

Btw. Java Upgrade was a hassle as my CLI showed 1.7 even after Java JDK install. I had to change my ~/.bash_profile (on OSX):

    export JAVA_HOME=$(/usr/libexec/java_home -v 1.8)

and needed to set a Cordova ENV var like this to make Cordova use the current Gradle:

    export CORDOVA_ANDROID_GRADLE_DISTRIBUTION_URL='http\://services.gradle.org/distributions/gradle-2.14.1-all.zip'

thanks!
